client.go

```diff
@@ -143,6 +150,39 @@ func (c *client) connection() (*sftp.Client, error) {
 	return c.client, nil
 }
 
+// trapConnectionError accepts any error from a call involving the SSH/SFTP connection.
+// If an error is encountered that causes either connection (SSH or SFTP) to be
+// lost it will tear down the connections. The next invocation of c.connection()
+// will re-establish new connections.
+//
+// When the error is captured by trapConnectionError the client will attempt to reconnect
+// and that new connection error will be returned.
+func (c *client) trapConnectionError(err error) error {
```

Comment:
Getting nit-picky, but I wonder if the naming could be more descriptive. 
* `handleConnectionError`
* `reconnectOnError`
* `tryReconnect`

Reply:
I renamed it to `clearConnectionOnError` which I hope is more clear.